internal/providers/azure/pke/workflow/azure.go

```diff
@@ -50,43 +50,54 @@ func (f *AzureClientFactory) New(organizationID uint, secretID string) (*pkgAzur
 	return cc, nil
 }
 
-func getOwnedTag(clusterName string) (string, string) {
-	return getOwnershipTagKey(clusterName), "owned"
+type Tag struct {
+	Key   string
+	Value string
 }
 
-func getSharedTag(clusterName string) (string, string) {
-	return getOwnershipTagKey(clusterName), "shared"
+func getOwnedTag(clusterName string) Tag {
+	return Tag{
+		Key:   getOwnershipTagKey(clusterName),
+		Value: "owned",
+	}
+}
+
+func getSharedTag(clusterName string) Tag {
+	return Tag{
+		Key:   getOwnershipTagKey(clusterName),
+		Value: "shared",
+	}
 }
 
 func getOwnershipTagKey(clusterName string) string {
 	return fmt.Sprintf("kubernetesCluster-%s", clusterName)
 }
 
-func tagsFrom(key, value string) map[string]string {
-	return map[string]string{
-		key: value,
+func tagsToMap(tags ...Tag) map[string]string {
+	m := make(map[string]string, len(tags))
+	for _, tag := range tags {
+		m[tag.Key] = tag.Value
 	}
+	return m
 }
 
 func HasOwnedTag(clusterName string, tags map[string]string) bool {
-	key, val := getOwnedTag(clusterName)
-	return hasTag(tags, key, val)
+	return hasTag(tags, getOwnedTag(clusterName))
 }
 
 func HasSharedTag(clusterName string, tags map[string]string) bool {
-	key, val := getSharedTag(clusterName)
-	return hasTag(tags, key, val)
+	return hasTag(tags, getSharedTag(clusterName))
 }
 
 func RemoveSharedTag(tags map[string]string, clusterName string) map[string]string {
```

Comment:
shouldn't this either work on a copy of the input map, or have no return value? maps are not like slices in this regard

Reply:
returning the modified map makes the function "chainable", e.g.: `tags = RemoveSomeOtherTag(RemoveSharedTag(tags, "my-cluster"), "my-other-key")`